android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletNftAdapter.java

```diff
@@ -0,0 +1,142 @@
+        return new ViewHolder(walletCoinView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        WalletListItemModel walletListItemModel = walletListItemModelList.get(position);
+        // When ViewHolder is re-used, it has the obeservers which are fired when
+        // we modifying checkbox. This may cause unwanted modifying of the model
+        holder.resetObservers();
+
+        holder.titleText.setText(walletListItemModel.getTitle());
+
+        PortfolioModel.NftDataModel nftDataModel = walletListItemModel.getNftDataModel();
+        holder.subTitleText.setText(nftDataModel.token.name);
+
+        holder.itemView.setOnClickListener(v -> {
+            onWalletListItemClick.onAssetClick(walletListItemModel.getBlockchainToken());
+        });
+
+        if (walletListItemModel.getBlockchainToken() == null
+                || !walletListItemModel.getBlockchainToken().logo.isEmpty()) {
+            Utils.setBitmapResource(mExecutor, mHandler, context, walletListItemModel.getIconPath(),
+                    walletListItemModel.getIcon(), holder.iconImg, null, true);
+        } else {
+            if (walletListItemModel.hasNftImageLink()
+                    && ImageLoader.isSupported(nftDataModel.nftMetadata.mImageUrl)) {
+                String url = nftDataModel.nftMetadata.mImageUrl;
+                ImageLoader.downloadImage(url, context, false, holder.iconImg, null);
+            } else {
+                Utils.setBlockiesBitmapCustomAsset(mExecutor, mHandler, holder.iconImg,
+                        walletListItemModel.getBlockchainToken().contractAddress,
+                        walletListItemModel.getBlockchainToken().symbol,
+                        context.getResources().getDisplayMetrics().density, null, context, false,
+                        (float) 0.9);
+            }
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return walletListItemModelList.size();
+    }
+
+    public void setWalletListItemModelList(List<WalletListItemModel> walletListItemModelList) {
+        this.walletListItemModelList = walletListItemModelList;
+    }
+
+    public void setOnWalletListItemClick(OnWalletListItemClick onWalletListItemClick) {
+        this.onWalletListItemClick = onWalletListItemClick;
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    public void removeItem(WalletListItemModel item) {
+        walletListItemModelList.remove(item);
+        notifyDataSetChanged();
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    public void addItem(WalletListItemModel item) {
+        walletListItemModelList.add(item);
+        notifyDataSetChanged();
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    public void clear() {
+        walletListItemModelList.clear();
+        notifyDataSetChanged();
+    }
+
+    public static class ViewHolder extends RecyclerView.ViewHolder {
```

Comment:
nit: can be private?

Reply:
It can't be private otherwise the RecyclerView will throw an exception.

![Screenshot from 2023-03-27 14-14-09](https://user-images.githubusercontent.com/3308503/227945808-f32ffbee-6a2f-4ed5-9ef7-bd354eb4a237.png)


It can be made package private